pointers/pointers.c

```diff
@@ -66,7 +97,21 @@ int string_compare(char *m, char *n)
 */
 char *find_string(char *haystack, char *needle)
 {
-
+    int i = 0, j = 0;
+    while ((*(haystack + j) != '\0')&&(*(needle + i) != '\0')) {
+        if (*(needle + i) != *(haystack + j)) {
+            j++;
+            i = 0;
+        }
+        else {
+            i++;
+            j++;
+        }
+    }
+    if (*(needle + i) == '\0')
+        return 1;
+    else
+        return -1;
 }
```

Comment:
My super old C99 compiler is giving all kinds of errors here, and I'm wondering if it's because you're returning 1 or -1 when it is a char function??